.travis.yml

```diff
@@ -71,86 +71,53 @@ matrix:
           - g++-7
           - python3.7-dev
           - python3.7-venv
+  - os: osx
+    osx_image: xcode11.5
+    env: PYTHON=3.7 CPP=14 COMPILER=clang
+    name: MacOS Python 3.7, c++14, Apple Clang
 
+#  - os: windows
+#    name: Windows Python 3.7, c++14, Visual Studio
+#    env: PYTHON=3.7 CPP=14 COMPILER=msvc
+
+env:
+  - DEBUG_CI=true
+    
 cache:
   directories:
   - $HOME/.local/bin
   - $HOME/.local/lib
   - $HOME/.local/include
   - $HOME/Library/Python
+
 before_install:
-- |
-  # Configure build variables
-  set -ex
-  if [ -n "$CLANG" ]; then
-    export CXX=clang++-$CLANG CC=clang-$CLANG
-    EXTRA_PACKAGES+=" clang-$CLANG llvm-$CLANG-dev"
-  else
-    EXTRA_PACKAGES+=" g++-$GCC"
-    export CXX=g++-$GCC CC=gcc-$GCC
-  fi
-  if [ -n "$CPP" ]; then CPP=-std=c++$CPP; fi
-  if [ "${PYTHON:0:1}" = "3" ]; then PY=3; fi
-  CMAKE_EXTRA_ARGS+=" -DBUILD_TESTS=ON"
-  if [ "$DEBUG" = true ]; then CMAKE_EXTRA_ARGS+=" -DCMAKE_BUILD_TYPE=Debug"; fi
-  if [ "$NOWND" = true ]; then CMAKE_EXTRA_ARGS+=" -DCARMA_DONT_REQUIRE_OWNDATA"; fi 
-  if [ "$SONAR" = true ]; then
-    export SONAR_SCANNER_VERSION=4.2.0.1873
-    export SONAR_SCANNER_HOME=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux
-    curl --create-dirs -sSLo $HOME/.sonar/sonar-scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_SCANNER_VERSION-linux.zip 
-    unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
-    export PATH=$SONAR_SCANNER_HOME/bin:$PATH
-    export SONAR_SCANNER_OPTS="-server"
-    
-    curl --create-dirs -sSLo $HOME/.sonar/build-wrapper-linux-x86.zip https://sonarcloud.io/static/cpp/build-wrapper-linux-x86.zip
-    unzip -o $HOME/.sonar/build-wrapper-linux-x86.zip -d $HOME/.sonar/
-    export PATH=$HOME/.sonar/build-wrapper-linux-x86:$PATH
-  fi
-  PY_CMD=python$PYTHON
-  curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-  ${PY_CMD} get-pip.py
-  set +ex
+  # Add reequirement for installation (Python)
+  - ${TRAVIS_BUILD_DIR}/.travis-ci/before_install.sh
+
 install:
-- |
   # Install dependencies
-  set -ex
-  if [ "$CLANG" = "7" ]; then
-    export CXXFLAGS="-stdlib=libc++"
-  fi
-  ${PY_CMD} -m pip install pip --upgrade
-  ${PY_CMD} -m pip install pytest numpy scipy cmake --upgrade 
-  set +ex
-- |
-  # CMake Configuration
-  set -ex
-  mkdir build
-  cd build
-  cmake ${CMAKE_EXTRA_ARGS} \
-    -DPYBIND11_PYTHON_VERSION=$PYTHON \
-    -DPYBIND11_CPP_STANDARD=$CPP \
-    ..
-  if [ "$SONAR" = true ]; then
-    cd ..
-    build-wrapper-linux-x86-64 --out-dir bw-output cmake --build build/ --config Release
-    sonar-scanner \
-      -Dsonar.organization=rurlus \
-      -Dsonar.projectKey=RUrlus_carma \
-      -Dsonar.sources=include/.,tests/src/. \
-      -Dsonar.host.url=https://sonarcloud.io \
-      -Dsonar.cfamily.build-wrapper-output=bw-output \
-      -Dsonar.login=890640f8296b91c98ccfa8ec2545220d68a4edb6
-  else
-    make
-    cd ..
-  fi
-  set +ex
+  - |
+    #!/usr/bin/env bash
```

Comment:
she-bang is required for macOS since default shell is zsh (to prevent possible incompatibilities)